### PR TITLE
Add drf opportunities

### LINKED ISSUE #220: Create Backend Schema in DRF
### Related Issue (Can be worked on in tandem)

- [x] #159 

Note: This issue can be done concurrently with its dependency.

### Overview

As part of our product we need to have a backend schema. For this issue we will add our tentative schema into our backend via DRF and a swagger UI.

### Action Items

- [x] See the db diagram below
- [x] Implement it using DRF
- [x] After implementation, add a swagger UI

### Resources/Instructions

[Resources](https://github.com/hackforla/CivicTechJobs/wiki/Resources)

Code for https://dbdiagram.io/d

```sql
// Creating tables
Table opportunities {
  id int [pk, increment] // auto-increment
  poster int
  job_title varchar
  project int
  max_commitment int [note: 'hours per week']
  min_commitment int [note: 'hours per week']
  max_duration int [note: 'days total']
  min_duration int [note: 'days total']
  positions_available int
  description varchar
  posted_date datetime [default: `now()`]
  role int
  experience int 
  required_tech int 
  preferred_tech int
}
```

### PR DESCRIPTION
**Pull request by Enzyme3**
Fixes #220 

Apologies... not sure why all the previous commits are listed out here - tried squashing them but doesn't seem to work. I'll try to fix it before raising any other PRs.

### Changes
- added backend schema 
- created /opportunities endpoint to list all opportunities
- re: tools/technologies, I just renamed the attribute/table to `skills`
  - do we need to split out the languages/tools/tech individually? I feel like that's an arbitrary distinction as regardless of what it is, an opportunity will list it and specify whether it is a required or preferred skill to have. LMK if otherwise

```
# sample response
[
    {
        "id": "1ad0916e-89f2-40d7-8d5a-1713a812a32b",
        "skills": [
            {
                "id": "92df2b94-ed1e-44cc-a664-4f80149c39f0",
                "is_required": true,
                "name": "Python"
            },
            {
                "id": "c56a5f44-fcbf-45e9-a14e-e2d672fc7080",
                "is_required": false,
                "name": "Django"
            }
        ],
        "experience_level": "Trained",
        "title": "Backend developer needed for super fun oppp!",
        "description": "come build some backends with us in django",
        "poster": "[123@hackforla.org](mailto:123@hackforla.org)",
        "hours_per_week": 8,
        "num_positions_available": 2,
        "posted_date": "2022-05-12",
        "project": {
            "id": "e080d5be-52d8-4dd4-aff5-8ac5f5d21c84",
            "name": "CivicTechJobs",
            "description": "CivicTechJobs helps technology practitioners interested in having civic impact to find job opportunities from a central hub of listings",
            "recurring_events": [
                {
                    "name": "CivicTechJobs Weekly All Hands",
                    "day_of_week": "Wednesday",
                    "start_time": "17:00:00",
                    "duration_in_minutes": 60,
                    "roles": [
                        {
                            "id": "36a18961-101e-4808-8f7d-24f143fcbb27",
                            "name": "All"
                        }
                    ]
                },
                {
                    "name": "CivicTechJobs Developer Meeting",
                    "day_of_week": "Wednesday",
                    "start_time": "18:00:00",
                    "duration_in_minutes": 30,
                    "roles": [
                        {
                            "id": "eb4ca87e-a8d0-4118-b4c2-22b8eb5d86c2",
                            "name": "Backend Developer"
                        },
                        {
                            "id": "6e8711bf-6e24-4e6c-b2c2-e542e8616324",
                            "name": "Frontend Developer"
                        }
                    ]
                },
                {
                    "name": "CivicTechJobs UI/UX Meeting",
                    "day_of_week": "Wednesday",
                    "start_time": "18:00:00",
                    "duration_in_minutes": 45,
                    "roles": [
                        {
                            "id": "8ab6bd23-3e55-436a-b678-2f9649487cd5",
                            "name": "UI/UX Designer"
                        }
                    ]
                }
            ]
        },
        "role": {
            "id": "eb4ca87e-a8d0-4118-b4c2-22b8eb5d86c2",
            "name": "Backend Developer"
        }
    },
...
```

To see in action, bring up app via `docker compose` and access endpoint via `http://localhost:8000/api/v1/opportunitites/`

To load sample data, run `docker compose run django python manage.py loaddata test_data.yaml`

### CODE REVIEW

**Comment by Enzyme3** [comment]
not sure why the linting is failing
Error #1 points to a linter.yml file
And error #2 points to a css file

neither has been touched/modified. Were some extra linting rules enabled?

**Comment by Aveline-art** [comment]
@Enzyme3 Feel free to ignore the linting error for now. It probably resulted from the way the branching was done.

File: app/config/settings.py [hunk 1]
```diff
@@ -45,15 +45,15 @@
...
-]
+
+MIDDLEWARE = [
+    'django.middleware.security.SecurityMiddleware',
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
```
Comment (Aveline-art) [inline app/config/settings.py:56]: This change broke the Django server.

**Review by Aveline-art — COMMENTED** [review]
I'll need some time to do a full review, but here's a quick note you might want to take a look at.

File: app/config/settings.py (hunk 1 above)
Comment (Enzyme3) [reply app/config/settings.py]: sorry, looks like the `]` somehow got dropped. Added it, and re-ran the linter which also changed the single quotes

**Review by Aveline-art — APPROVED** [review]
This looks good to me! Thanks for working on this!

**Pull request merged**